### PR TITLE
feat: call multicall v3 contract, add unit test for amoy chain

### PR DESCRIPTION
**Pull request by galvin96**

### CODE REVIEW

File: test/amoy.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { ethers } from 'ethers';
+import { assert } from 'chai';
+import { Contract, Provider } from '../src';
+
+const rpcUrl = process.env.RPC_URL
+const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
+const ethcallProvider = new Provider(provider, 80002);
+
+it('human readable abi', async () => {
```
Comment (thi-investax) [inline test/amoy.test.ts]: the test name is not relevant

File: src/provider.ts [hunk 2]
```diff
@@ -33,22 +33,19 @@ export class Provider {
...
+  97: '0xcA11bde05977b3631167028862bE2a173976CA11',
+  100: '0xcA11bde05977b3631167028862bE2a173976CA11',
+  128: '0xcA11bde05977b3631167028862bE2a173976CA11',
+  137: '0xcA11bde05977b3631167028862bE2a173976CA11',
+  250: '0xcA11bde05977b3631167028862bE2a173976CA11',
+  42161: '0xcA11bde05977b3631167028862bE2a173976CA11',
+  43114: '0xcA11bde05977b3631167028862bE2a173976CA11',
+  80001: '0xcA11bde05977b3631167028862bE2a173976CA11',
+  80002: '0xcA11bde05977b3631167028862bE2a173976CA11',
+  84532: '0xcA11bde05977b3631167028862bE2a173976CA11',
```
Comment (thi-investax) [inline src/provider.ts]: missing the chain id for base mainnet

File: package.json [hunk 3]
```diff
@@ -20,12 +20,14 @@
...
     "@types/chai": "^4.1.7",
     "@types/mocha": "^5.2.5",
+    "@types/node": "^20.14.10",
     "chai": "^4.2.0",
     "mocha": "^5.2.0",
     "tslint": "^5.12.0",
     "typescript": "^4.2.4"
   },
   "dependencies": {
+    "dotenv": "^16.4.5",
```
Comment (thi-investax) [inline package.json]: @galvin96 , i wonder why we need `dotenv` here?

Comment (galvin96) [reply package.json]: In this file: [test/amoy.test.ts](https://github.com/IX-Swap/ethers-multicall/pull/1/files/00715a11f46bc54349397057cfe92ae06669d808#diff-8649aaa272cb5cbd432f8b2c9c4fb62168338288bb5eaa6b3253007ccf137c59) we need to retrieve the RPC url from the .env file. Thus, I added this package.

Comment (thi-investax) [reply package.json]: when we distribute npm package, shouldn't  include irrelevant packages that serves chores or testing. 
https://investax.atlassian.net/wiki/spaces/IXS/pages/263585796/Development+guidelines+Front-end+-+React#Libraries%2Fdependencies

There are 2 ways I can propose to make sure we follow a good practice:

1. Put dotenv in dev Depdencies
2. Harcode the rpc in the test file. 

I prefer the second, if someone else need to maintain this project in the future. they don't need to go to the internet and copy the rpc urls to execute these test files. Everything is already available

**Comment by thi-investax** [comment]
@galvin96 could we have test cases for base, polygon please? just want to make sure this should work on the mainnet when we deploy to production

File: test/amoy.test.ts (hunk 1 above)
Comment (galvin96) [reply test/amoy.test.ts]: updated.

File: src/provider.ts (hunk 2 above)
Comment (galvin96) [reply src/provider.ts]: updated.

File: package.json (hunk 3 above)
Comment (galvin96) [reply package.json]: Went with the 2nd approach. updated.

**Review by thi-investax — APPROVED** [review]

**Review by thi-investax — APPROVED** [review]

**Pull request merged**